src/shared/hooks/use-draggable-list-modal.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button, ButtonProps } from '@viaa/avo2-components';
+import React, { ReactNode, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import DraggableBlock from '../components/DraggableBlock/DraggableBlock';
+import DraggableListModal, {
+	DraggableListModalProps,
+} from '../components/DraggableList/DraggableListModal';
+
+export function useDraggableListModal(config?: {
+	button?: Partial<ButtonProps>;
+	modal?: Partial<DraggableListModalProps>;
+}): [ReactNode, ReactNode] {
+	const [t] = useTranslation();
+	const [isOpen, setIsOpen] = useState<boolean>(false);
+
+	const button = (
+		<Button
+			icon="shuffle"
+			{...config?.button}
+			type="secondary"
+			label={t('collection/components/collection-or-bundle-edit___herorden-fragmenten')}
+			title={t(
+				'collection/components/collection-or-bundle-edit___herorden-de-fragmenten-via-drag-and-drop'
+			)}
+			onClick={(e) => {
+				setIsOpen(true);
+				config?.button?.onClick?.(e);
+			}}
+		/>
+	);
+
+	const modal = (
+		<DraggableListModal
+			{...config?.modal}
+			renderItem={(item) => <DraggableBlock block={item} />}
+			isOpen={isOpen}
+			onClose={(update?: any[]) => {
+				setIsOpen(false);
+				config?.modal?.onClose?.(update);
+			}}
+		/>
+	);
+
+	return [button, modal];
```

Comment:
optional: modals mogen in principe gelijk waar gerenderd worden
Dus kunnen we deze hook mogelijks omvormen naar een component die dan de button en de modal op 1 plaats renderd.

```
  <DraggableListModalWithButton config={} />
```

in plaats van:
```
const [draggableListButton, draggableListModal] = useDraggableListModal({})

...

{draggableListButton}

...

{draggableListModal}
```

Reply:
Not sure how I'd call this kind of component, `DraggableList` & `DraggableListModal` are already in use by the sub-components. Not doing for now.

Reply:
DraggableListModalWithButton